smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,676 @@
+        job.entity.type,
+        "timestep",
+        exp_dir,
+        logger,
+    )
+
+
+class ManifestEventHandler(PatternMatchingEventHandler):
+    """The ManifestEventHandler monitors an experiment for changes and updates
+    a telemetry datastore as needed.
+
+    It contains event handlers that are triggered by changes to a runtime experiment
+    manifest. The runtime manifest differs from a standard manifest. A runtime manifest
+    may contain multiple experiment executions in a `runs` collection.
+
+    It also contains a long-polling loop that checks experiment entities for updates
+    at each timestep."""
+
+    def __init__(
+        self,
+        pattern: str,
+        logger: logging.Logger,
+        ignore_patterns: t.Any = None,
+        ignore_directories: bool = True,
+        case_sensitive: bool = False,
+    ) -> None:
+        super().__init__(
+            [pattern], ignore_patterns, ignore_directories, case_sensitive
+        )  # type: ignore
+        self._logger = logger
+        self._tracked_runs: t.Dict[int, Run] = {}
+        self._tracked_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._completed_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._launcher_type: str = ""
+        self._launcher: t.Optional[Launcher] = None
+        self.job_manager: JobManager = JobManager(threading.RLock())
+        self._launcher_map: t.Dict[str, t.Type[Launcher]] = {
+            "slurm": SlurmLauncher,
+            "pbs": PBSLauncher,
+            "cobalt": CobaltLauncher,
+            "lsf": LSFLauncher,
+            "local": LocalLauncher,
+        }
+
+    def init_launcher(self, launcher: str) -> Launcher:
+        """Initialize the controller with a specific type of launcher.
+        SmartSim currently supports slurm, pbs(pro), cobalt, lsf,
+        and local launching
+
+        :param launcher: which launcher to initialize
+        :type launcher: str
+        :raises SSUnsupportedError: if a string is passed that is not
+                                    a supported launcher
+        :raises TypeError: if no launcher argument is provided.
+        """
+
+        if launcher is not None:
+            launcher = launcher.lower()
+            if launcher in self._launcher_map:
+                return self._launcher_map[launcher]()
+            raise ValueError("Launcher type not supported: " + launcher)
+
+        raise TypeError("Must provide a 'launcher' argument")
+
+    def set_launcher(self, launcher_type: str) -> None:
+        """Set the launcher for the experiment"""
+        if launcher_type != self._launcher_type:
+            self._launcher_type = launcher_type
+            self._launcher = self.init_launcher(launcher_type)
+            self.job_manager.set_launcher(self._launcher)
+            self.job_manager.add_job_onstart_callback(track_started)
+            self.job_manager.add_job_onstop_callback(track_completed)
+            self.job_manager.add_job_onstep_callback(track_timestep)
```

Comment:
there was an issue i had to fix yesterday w/ the watch directory & the watch pattern (if dir didn't exist, the watch configuration would fail). swapping to the exp_dir and using the relative path to the manifest in the pattern fixed this. let's retest together to be sure you see what i do